packages/material-ui/src/ButtonBase/ButtonBase.test.js

```diff
@@ -1,23 +1,22 @@
 import React from 'react';
+import { act } from 'react-dom/test-utils';
 import { assert } from 'chai';
-import { spy, useFakeTimers } from 'sinon';
+import { spy } from 'sinon';
 import rerender from 'test/utils/rerender';
 import {
   createShallow,
   createMount,
   describeConformance,
   getClasses,
-  unwrap,
 } from '@material-ui/core/test-utils';
 import TouchRipple from './TouchRipple';
 import ButtonBase from './ButtonBase';
 import consoleErrorMock from 'test/utils/consoleErrorMock';
 import * as PropTypes from 'prop-types';
 
-const ButtonBaseNaked = unwrap(ButtonBase);
-
 function focusVisible(element) {
-  element.ownerDocument.dispatchEvent(new window.Event('keydown'));
+  element.blur();
```

Comment:
Why do we call blur? Shouldn't focus be enough?

Reply:
If you call focus repeatedly on a DOM node it won't dispatch a focus event. JSDOM does not implement this behavior correctly. If you remove this line JSDOM tests will pass but test:karma will fail.